assets/src/edit-story/components/library/panes/pageTemplates/templateSave.js

```diff
@@ -80,38 +99,80 @@ const StyledText = styled(Text)`
   margin-bottom: 8px;
 `;
 
-const StyledButton = styled(Button)`
-  height: 32px;
-`;
+function TemplateSave({ pageSize, setShowDefaultTemplates, updateList }) {
+  const {
+    actions: { addPageTemplate },
+  } = useAPI();
+  const { showSnackbar } = useSnackbar();
+
+  const { currentPage } = useStory(({ state: { currentPage } }) => ({
+    currentPage,
+  }));
 
-function TemplateSave({ pageSize }) {
   const { customPageTemplates } = useFeatures();
-  const handleSaveTemplate = useCallback(() => {}, []);
+  const handleSaveTemplate = useCallback(() => {
+    // Don't add empty page.
+    if (isDefaultPage(currentPage)) {
+      showSnackbar({
+        message: __(
+          'An empty page can’t be saved as a template. Add elements and try again.',
+          'web-stories'
+        ),
+        dismissable: true,
+      });
+      return;
+    }
+    addPageTemplate({ ...currentPage, id: uuidv4(), title: null }).then(
+      (addedTemplate) => {
+        updateList?.(addedTemplate);
+        // @todo Add error handling.
```

Comment:
Why is error handling in a TODO here?

Reply:
This issue doesn't include error handling yet, will be done in a next issue (still more to come for page templates before enabling the flag)